week1/index.js

```diff
@@ -16,10 +92,12 @@ server.listen(port, function(error) {
 });
 
 // Create a event handler for "request"
-// this is an alternative way
 server.on('request', function(request, response) {
-  console.log('New http request received', request.url);
-  response.setHeader('content-type', 'text/html');
-  response.write('<html><head></head><body><h1>Hello world</h1></body></html>');
-  response.end();
-});
\ No newline at end of file
+    //send the URL to be checked for the prober response
+    //send the basic respons to the checkInput function to modify it
+    let inputToReturn = checkInput(request.url,responsToSendBack);
+    response.statusCode = inputToReturn.statusCode;
+    response.setHeader('content-type', inputToReturn.type);
+    response.write(inputToReturn.code);
+    response.end();
```

Comment:
This is a very well-designed listener handler - it's clear what the order of events are, and that all the content processing is done by another function. Great job. 😀